Prog/2019/Internet Access.sas

```diff
@@ -0,0 +1,183 @@
+ Version:  SAS 9.4
+ Environment:  Local Windows session
+ 
+ Description:  
+
+**************************************************************************/
+
+%include "L:\SAS\Inc\StdLocal.sas";
+
+** Define libraries **;
+%DCData_lib( ACS )
+%DCData_lib (RegHsg);
+
+%let year= 2013_17;
+
+proc format;
+  value Jurisdiction
+	0= "Total"
+    1= "DC"
+	2= "Charles County"
+	3= "Frederick County "
+	4="Montgomery County"
+	5="Prince Georges "
+	6="Arlington"
+	7="Fairfax, Fairfax city and Falls Church"
+	8="Loudoun"
+	9="Prince William, Manassas and Manassas Park"
+    10="Alexandria"
+  	;
+run; 
+
+data InternetAcccess ;
+	  set acs.Acs_2013_17_va_sum_tr_tr10 acs.Acs_2013_17_dc_sum_tr_tr10 acs.Acs_2013_17_md_sum_tr_tr10 acs.Acs_2013_17_wv_sum_tr_tr10;
+	      
+	  keep geo2010 geoid totpop_&year. medfamincm_&year. popalonew_&year. popaloneb_&year. popaloneh_&year. popalonea_&year. popaloneaiom_&year. 
+	       Numdialup_&year. Numbroadbandall_&year. Numcellular_&year. Numcellularonly_&year. Numbroadband_&year. Numbroadbandonly_&year. Numsatellite_&year.
+           Numsatelliteonly_&year. Numotheronly_&year. Numaccesswosub_&year. Numnointernet_&year. tothhdemom pctinternet percent100Kplus percentunder15K
+           percent15to50K percent50to75K percent75Kto100K county metro15 Jurisdiction hshldinc100000plus_2013_17 hshldincunder15000_2013_17 hshldinc15000to34999_2013_17 hshldinc35000to49999_2013_17
+           hshldinc50000to74999_2013_17 hshldinc75000to99999_2013_17 pctinternet pctnointernet Numwithinternet_2013_17;
+
+      county= substr(geo2010, 1,5);
+	  ucounty=county;
+	  geoid= geo2010;
+
+%ucounty_jurisdiction
+
+	  metro15 = put( county, $ctym15f. );
+      tothhdemom= hshldinc100000plus_2013_17+ hshldincunder15000_2013_17+ hshldinc15000to34999_2013_17+ hshldinc35000to49999_2013_17+
+      hshldinc50000to74999_2013_17+ hshldinc75000to99999_2013_17;
+      percent100Kplus= hshldinc100000plus_2013_17/tothhdemom;
+	  percentunder15K= hshldincunder15000_2013_17/tothhdemom;
+	  percent15to50K= (hshldinc15000to34999_2013_17+ hshldinc35000to49999_2013_17)/tothhdemom;
+	  percent50to75K= hshldinc50000to74999_2013_17/tothhdemom;
+      percent75Kto100K= hshldinc75000to99999_2013_17/tothhdemom;
+	  internetdenom= Numdialup_&year. + Numbroadbandall_&year. + Numcellular_&year. + Numcellularonly_&year. + Numbroadband_&year. + Numbroadbandonly_&year. + Numsatellite_&year.+ 
+                     Numsatelliteonly_&year. + Numotheronly_&year. + Numaccesswosub_&year. + Numnointernet_&year.;
+	  pctinternet= (internetdenom- Numnointernet_&year.)/internetdenom;
+	  pctnointernet= 1- pctinternet;
+
+	  format Jurisdiction Jurisdiction. ;
+run;
+
+proc export data= InternetAcccess
+   outfile="&_dcdata_default_path\Requests\Prog\2019\Internet_access_tract.csv"
+   dbms=csv
+   replace;
+run;
+
+data BroadbandAcccess ;
+	  set acs.Acs_2013_17_va_sum_tr_tr10 acs.Acs_2013_17_dc_sum_tr_tr10 acs.Acs_2013_17_md_sum_tr_tr10 acs.Acs_2013_17_wv_sum_tr_tr10;
```

Comment:
you should in one data step read in all the variables you need from the ACS and create the jurisdictions. Not do it repeatedly.